### DISCUSSION TITLE
What about CBOR?

### DISCUSSION THREAD

**Discussion by bryceschober** [stephenberry/glaze#783] [Ideas]
@stephenberry What do you think about implementing support for [CBOR](https://cbor.io/)? I know that you probably designed & implemented BEVE partially because of CBOR's limitations, but now that CBOR is going to be used for more standardized IOT-related protocols, including for [standardizing firmware update packages (SUIT)](https://datatracker.ietf.org/wg/suit/about/) via the [CBOR Object Signing and Encryption (COSE) standard](https://datatracker.ietf.org/doc/html/rfc9052)...

Anyway, I wonder what the level of complexity would be in implementing serialization for `cbor` alongside your `s/binary/beve/` implementation?

**Comment by stephenberry** [comment]
@bryceschober, I've used CBOR a bit and studied the specification, but I don't currently have a use for it and have more pressing developments for Glaze.

However, if someone wanted to contribute to Glaze with CBOR support, I would happily add them to the Glaze team and appreciate the additional format.

CBOR does have a number of downsides, especially when it comes to performance, which is why I've gone with BEVE. But, I understand its use as a standard and I think it makes sense for Glaze because of its close relation to JSON.

A CBOR implementation in Glaze would likely be extremely high performing versus current implementations.

**Reply by bryceschober** [reply]
So @stephenberry do you have any guess as to the complexity would be in implementing serialization for `cbor` alongside your `s/binary/beve/` implementation? Would the `beve` support be the best baseline for the points to hit for CBOR support?

**Reply by stephenberry** [reply]
@bryceschober, the beve code would provide you with the glaze architecture (see from_binary and to_binary specializations). You would then need to refer to the CBOR specification for the actual binary layout.

If you wanted to start out, just try implementing reading/writing booleans or integers.
